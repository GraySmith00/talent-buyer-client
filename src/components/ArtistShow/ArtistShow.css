.artist-show {
  display: flex;
  background-color: #21222b;
  color: #fff;
  overflow: scroll; }
  .artist-show .artist-name {
    text-decoration: underline;
    text-decoration-color: #C73B96; }
  .artist-show .back-button {
    width: 3em; }
  .artist-show .bio {
    font-size: 1.45em; }
  .artist-show .button {
    padding-left: 1em; }
  .artist-show .genre {
    color: #D52F7F;
    padding-left: 1em; }
  .artist-show .genre-wrap {
    align-items: center;
    display: flex;
    font-size: 1.2em;
    width: 100%; }
  .artist-show .followers {
    color: #D52F7F;
    padding-left: 1em;
    padding-right: 1em; }
  .artist-show .followers-wrap {
    align-items: center;
    display: flex;
    font-size: 1.2em; }
  .artist-show .main-content {
    background-color: #131621;
    border-radius: 1.5rem;
    display: flex;
    flex-direction: column;
    height: 40em;
    justify-content: space-evenly;
    margin: auto;
    padding: 1em;
    width: 60em; }
  .artist-show .main-content-frame {
    align-items: center;
    background-color: #C73B96;
    border-radius: 1.5rem;
    display: flex;
    justify-content: center;
    height: 40.1em;
    margin: auto;
    width: 60.1em; }
  .artist-show .name-wrap {
    align-items: center;
    display: flex;
    justify-content: space-between;
    width: 96%; }
  .artist-show .portrait-wrap {
    display: flex;
    width: 100%; }
  .artist-show .profile-image {
    border-radius: 50%;
    width: 15em; }
  .artist-show .profile-frame {
    align-items: center;
    background-color: #131621;
    border-radius: 50%;
    display: flex;
    height: 15.25rem;
    justify-content: center;
    width: 15.25rem; }
  .artist-show .similar {
    color: #D52F7F;
    padding-left: 1em; }
  .artist-show .similar-wrap {
    align-items: center;
    display: flex;
    font-size: 1.2em; }
  .artist-show .spotify-icon {
    width: 3em; }
  .artist-show .stat-wrap {
    display: flex;
    flex-direction: column;
    margin-left: 1em;
    width: 100%; }
