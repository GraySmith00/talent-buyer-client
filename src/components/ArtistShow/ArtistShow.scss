@import '../../styles/variables.scss';

.artist-show {
  background-color: #21222b;
  color: #fff;
  display: flex;
  overflow: scroll;

  .artist-name {
    text-decoration: underline;
    text-decoration-color: $pink-four;
  }

  .back-button {
    width: 3em;
  }

  .bio {
    font-size: 1em;
  }

  .button {
    padding-left: 1em;
  }

  .city-header {
    color: $pink-four;
    padding-left: 0.4em;
  }

  .genre {
    color: $pink-two;
    padding-left: 1em;
  }

  .genre-wrap {
    align-items: center;
    display: flex;
    font-size: 1.2em;
    width: 100%;
  }

  .followers {
    color: $pink-two;
    padding-left: 1em;
    padding-right: 1em;
  }

  .followers-wrap {
    align-items: center;
    display: flex;
    font-size: 1.2em;
  }

  .main-content {
    background-color: #131621;
    border-radius: 1.5rem;
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 47em;
    margin: auto;
    padding: 1.5em;
    width: 72em;
  }

  .main-content-frame {
    align-items: center;
    background-color: $pink-four;
    border-radius: 1.5rem;
    display: flex;
    justify-content: center;
    height: 47.3em;
    margin: auto;
    width: 72.3em;
  }

  .name-wrap {
    align-items: center;
    display: flex;
    justify-content: space-between;
    width: 96%;
  }

  .event-container {
    height: 140px;
    overflow: scroll;
    background-color: #32333e;
  }

  .portrait-wrap {
    display: flex;
    width: 100%;
  }

  .profile-image {
    border-radius: 50%;
    width: 15em;
  }

  .profile-frame {
    align-items: center;
    background-color: #131621;
    border-radius: 50%;
    display: flex;
    height: 15.25rem;
    justify-content: center;
    width: 15.25rem;
  }

  .similar {
    color: $pink-two;
    padding-left: 1em;
  }

  .similar-wrap {
    align-items: center;
    display: flex;
    font-size: 1.2em;
  }

  .spotify-icon {
    width: 3em;
  }

  .stat-wrap {
    display: flex;
    flex-direction: column;
    margin-left: 1em;
    width: 100%;
  }

  .recent-table-headings {
    display: grid;
    grid-template-columns: repeat(3, 10fr);
    border-bottom: 1px solid #fff;
    align-items: center;
    padding-left: 2rem;
    margin: 0;
    h3 {
      margin-bottom: 5px;
    }
  }

  .table-row {
    display: grid;
    grid-template-columns: repeat(3, 10fr);
    align-items: center;
    justify-content: center;
    margin-left: 2rem;
    font-weight: 300;
    font-size: 0.7rem;
  }
}
