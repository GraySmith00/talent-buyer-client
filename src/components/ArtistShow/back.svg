<?xml version="1.0" encoding="iso-8859-1"?>
<!-- Generator: Adobe Illustrator 18.1.1, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" id="Capa_1" x="0px" y="0px" viewBox="0 0 16 16" style="enable-background:new 0 0 16 16;" xml:space="preserve" width="512px" height="512px">
<g>
	<path d="M8,0C3.582,0,0,3.582,0,8s3.582,8,8,8s8-3.582,8-8S12.418,0,8,0z M8,14c-2.762,0-5-2.238-5-5   c0-0.615,0.111-1.204,0.314-1.749l1.71,1.368C5.008,8.744,5,8.871,5,9c0,1.657,1.343,3,3,3s3-1.343,3-3S9.657,6,8,6v2L4,5l4-3v2   c2.762,0,5,2.238,5,5S10.762,14,8,14z" fill="#c73b96"/>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
</svg>
