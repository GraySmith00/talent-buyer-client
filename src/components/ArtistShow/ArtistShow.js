import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { getArtist } from '../../Utils/backendApiCalls';
import { getLastFMInfo } from '../../Utils/lastFM/lastFMApiCalls';
import './ArtistShow.css';
import ArtistMediaPlayer from '../artistInfoCommon/ArtistMediaPlayer/ArtistMediaPlayer';
import Nav from '../Nav/Nav';
import LoadingSpinner from '../LoadingSpinner/LoadingSpinner';
import VenueHistory from '../VenueHistory/VenueHistory';

export class ArtistShow extends Component {
  state = {
    artist: {}
  };

  async componentDidMount() {
    const { id } = this.props;
    const artist = await getArtist(id);
    const lastFMInfo = await getLastFMInfo(artist.name);

    this.setState({
      artist: {
        ...artist,
        ...lastFMInfo
      }
    });
  }

  cleanName = name => {
    const trimSpace = name.replace(/ /g, '');
    return trimSpace;
  };

  render() {
    const { artist } = this.state;
    const { currentVenue } = this.props;
    const {
      name,
      image,
      bio,
      spotify_followers,
      genres,
      similarArtists,
      spotify_url
    } = this.state.artist;

    let displayArtist;

    if (!similarArtists || !genres) {
      displayArtist = <LoadingSpinner />;
    } else {
      displayArtist = (
        <Fragment>
          <section className="profile">
            <img className="profile-image" src={image} alt="artist-profile" />
            <div className="stat-wrap">
              <h1>{name}</h1>
              <span className="stat">
                <h3>Spotify Followers:</h3>
                <p>{spotify_followers}</p>
              </span>
              <span className="stat">
                <h3>Genres:</h3>
                <p>{genres.map(g => g.name).join(', ')}</p>
              </span>
              <span className="stat">
                <h3>Similar Artists:</h3>
                <p>{similarArtists.map(a => a.name).join(', ')}</p>
              </span>
              <div className="social-links">
                <a href={spotify_url} target="_blank" rel="noopener noreferrer">
                  <img src={require('./spotifyIcon.png')} alt="spotify" />
                </a>
                <a href={spotify_url} target="_blank" rel="noopener noreferrer">
                  <img src={require('./soundcloudIcon.png')} alt="spotify" />
                </a>
              </div>
            </div>
          </section>
          <section className="bio">
            <p>{bio}</p>
          </section>
          <section className="media-player">
            <ArtistMediaPlayer name={this.cleanName(name)} />
          </section>
          <section className="venue-history">
            <VenueHistory artist={artist} currentVenue={currentVenue} />
          </section>
        </Fragment>
      );
    }

    return (
      <div className="artist-show">
        <Nav />
        <div className="main-content">
          <Link to={`/artists`}>
            <i className="fas fa-arrow-left" />
          </Link>
          {displayArtist}
        </div>
      </div>
    );
  }
}

ArtistShow.propTypes = {
  currentVenue: PropTypes.object
};

export const mapStateToProps = state => ({
  currentVenue: state.currentVenue
});

export default connect(mapStateToProps)(ArtistShow);
