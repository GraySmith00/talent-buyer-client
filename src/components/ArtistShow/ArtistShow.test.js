import React from 'react';
import { shallow } from 'enzyme';
import { ArtistShow } from './ArtistShow';

jest.mock('../../Utils/backendApiCalls');

describe('ArtistShow component', () => {
  let wrapper;
  let mockId;

  beforeEach(() => {
    mockId = 4;
    wrapper = shallow(<ArtistShow id={mockId} />, {
      disableLifecycleMethods: true
    });
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
