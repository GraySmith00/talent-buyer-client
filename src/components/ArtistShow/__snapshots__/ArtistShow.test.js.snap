// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistShow component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArtistShow
    id={4}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Nav) />,
        <div
          className="main-content-frame"
>
          <div
                    className="main-content"
          >
                    <LoadingSpinner />
          </div>
</div>,
      ],
      "className": "artist-show",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="main-content"
>
            <LoadingSpinner />
</div>,
          "className": "main-content-frame",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <LoadingSpinner />,
            "className": "main-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Nav) />,
          <div
            className="main-content-frame"
>
            <div
                        className="main-content"
            >
                        <LoadingSpinner />
            </div>
</div>,
        ],
        "className": "artist-show",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="main-content"
>
              <LoadingSpinner />
</div>,
            "className": "main-content-frame",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <LoadingSpinner />,
              "className": "main-content",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
